third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -139,8 +140,16 @@ pub enum Type {
     Address,
     Signer,
     Vector(Box<Type>),
-    Struct(CachedStructIndex),
-    StructInstantiation(CachedStructIndex, Vec<Type>),
+    Struct {
+        index: CachedStructIndex,
+        ability: AbilitySet,
+    },
+    StructInstantiation {
+        index: CachedStructIndex,
+        ty_args: Vec<Type>,
```

Comment:
Vec<Box<Type>> To avoid deep (recursive) copies when Vec grows.

Reply:
It should of course be Vec<&Type> as the Type already exists for the appropriate lifetime, but gl convincing rust

Reply:
Could you elaborate? (I'm still rust beginner).

Wouldn't Vec<&Type> require the the elements of the vectors to be owned by some other data structure (or on the stack of the scope) that outlives the Vec?  I thought here the Vec is the owner of its elements.

Reply:
I'm fairly confident that types remain present for the duration of the loader